File: code/back-office/front/src/app/dto/pending-electoral-result.ts
export class PendingElectoralResult {
  id: number;
  electionId: number;
  election: string;
  pollingStationId: number;
  pollingStation: string;
  voters: number;
  maleUnder36: number;
  femaleUnder36: number;
  male36AndOver: number;
  female36AndOver: number;
  disabledPeople: number;
  visuallyImpairedPeople: number;
  blankVotes: number;
  nullVotes: number;
  details: {
    id: number;
    candidateRegistrationId: number;
    candidateId: number;
    candidateInformation: string;
    candidateImagePath: string;
    politicalEntityId: number;
    politicalEntity: string;
    politicalEntityDescription: string;
    votes: number;
  }[];
  images: {
    imagePath: string;
  }[];
  status: number;

  constructor(
    id: number,
    electionId: number,
    election: string,
    pollingStationId: number,
    pollingStation: string,
    voters: number,
    maleUnder36: number,
    femaleUnder36: number,
    male36AndOver: number,
    female36AndOver: number,
    disabledPeople: number,
    visuallyImpairedPeople: number,
    blankVotes: number,
    nullVotes: number,
    details: {
      id: number;
      candidateRegistrationId: number;
      candidateId: number;
      candidateInformation: string;
      candidateImagePath: string;
      politicalEntityId: number;
      politicalEntity: string;
      politicalEntityDescription: string;
      votes: number;
    }[],
    images: {
      imagePath: string;
    }[],
    status: number
  ) {
    this.id = id;
    this.electionId = electionId;
    this.election = election;
    this.pollingStationId = pollingStationId;
    this.pollingStation = pollingStation;
    this.voters = voters;
    this.maleUnder36 = maleUnder36;
    this.femaleUnder36 = femaleUnder36;
    this.male36AndOver = male36AndOver;
    this.female36AndOver = female36AndOver;
    this.disabledPeople = disabledPeople;
    this.visuallyImpairedPeople = visuallyImpairedPeople;
    this.blankVotes = blankVotes;
    this.nullVotes = nullVotes;
    this.details = details;
    this.images = images;
    this.status = status;
  }
}

File: code/back-office/server/src/main/java/ceni/system/votesync/model/view/election/result/PendingElectoralResult.java
package ceni.system.votesync.model.view.election.result;

import java.util.List;

import org.springframework.data.annotation.Immutable;

import ceni.system.votesync.model.entity.election.result.ResultImage;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.OneToMany;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "resultats_en_attente")
@Immutable
public class PendingElectoralResult {

    @Id
    private Integer id;

    @Column(name = "id_election")
    private Integer electionId;

    @Column(name = "nom_election")
    private String election;

    @Column(name = "id_bv")
    private Integer pollingStationId;

    @Column(name = "nom_bv")
    private String pollingStation;

    @Column(name = "identifiant_operateur_validateur")
    private String operatorIdentifier;

    @Column(name = "inscrits")
    private Integer voters;

    @Column(name = "homme_moins_36")
    private Integer maleUnder36;

    @Column(name = "femme_moins_36")
    private Integer femaleUnder36;

    @Column(name = "homme_36_plus")
    private Integer male36AndOver;

    @Column(name = "femme_36_plus")
    private Integer female36AndOver;

    @Column(name = "handicapes")
    private Integer disabledPeople;

    @Column(name = "malvoyants")
    private Integer visuallyImpairedPeople;

    @Column(name = "blancs")
    private Integer blankVotes;

    @Column(name = "nuls")
    private Integer nullVotes;

    @OneToMany(fetch = FetchType.EAGER)
    @JoinColumn(name = "id_resultat")
    private List<VResultDetails> details;

    @OneToMany(fetch = FetchType.EAGER)
    @JoinColumn(name = "id_resultat")
    private List<ResultImage> images;

    @Column(name = "etat")
    private Integer status;
}
